pandas/tests/frame/test_timeseries.py

```diff
@@ -323,6 +323,14 @@ def test_asfreq_datetimeindex(self):
         ts = df['A'].asfreq('B')
         tm.assertIsInstance(ts.index, DatetimeIndex)
 
+    def test_asfreq_datetimeindex_empty_series(self):
+        # GH 14340
+        expected = Series(index=pd.DatetimeIndex(
+            ["2016-09-29 11:00"])).asfreq('H')
+        result = Series(index=pd.DatetimeIndex(["2016-09-29 11:00"]),
+                        data=[3]).asfreq('H')
```

Comment:
Can you create one of the two without using `asfreq`.

Further, it is not fully clear what is what you actually testing, as the result of the empty series is called `expected` (you typically call the constructed expected value `expected` to compare the result of what you want to test)